hardware/ni_x_series_scanner.py

```diff
@@ -0,0 +1,1028 @@
+
+    !!!!!! NI USB 63XX, NI PCIe 63XX and NI PXIe 63XX DEVICES ONLY !!!!!!
+
+    See [National Instruments X Series Documentation](@ref nidaq-x-series) for details.
+
+    stable: Kay Jahnke, Alexander Stark
+
+    Example config for copy-paste:
+
+    nicard_6343:
+        module.Class: 'ni_x_series_scanner.NationalInstrumentsXSeriesScanner'
+        photon_sources:
+            - '/Dev1/PFI8'
+        #    - '/Dev1/PFI9'
+        default_scanner_clock_frequency: 100 # optional, in Hz
+        scanner_clock_channel: '/Dev1/Ctr2'
+        pixel_clock_channel: '/Dev1/PFI6'
+        scanner_ao_channels:
+            - '/Dev1/AO0'
+            - '/Dev1/AO1'
+            - '/Dev1/AO2'
+            - '/Dev1/AO3'
+        scanner_ai_channels:
+            - '/Dev1/AI1'
+        scanner_counter_channels:
+            - '/Dev1/Ctr3'
+        scanner_voltage_ranges:
+            - [-10, 10]
+            - [-10, 10]
+            - [-10, 10]
+            - [-10, 10]
+        scanner_position_ranges:
+            - [0e-6, 200e-6]
+            - [0e-6, 200e-6]
+            - [-100e-6, 100e-6]
+            - [-10, 10]
+
+        max_counts: 3e7
+        read_write_timeout: 10
+        counting_edge_rising: True
+
+    """
+
+    _modtype = 'NICard'
+    _modclass = 'hardware'
+
+    # config options
+    _photon_sources = ConfigOption('photon_sources', missing='error')
+
+    # confocal scanner
+    _default_scanner_clock_frequency = ConfigOption(
+        'default_scanner_clock_frequency', 100, missing='info')
+    _scanner_clock_channel = ConfigOption('scanner_clock_channel', missing='warn')
+    _pixel_clock_channel = ConfigOption('pixel_clock_channel', None)
+    _scanner_ao_channels = ConfigOption('scanner_ao_channels', missing='error')
+    _scanner_ai_channels = ConfigOption('scanner_ai_channels', [], missing='info')
+    _scanner_counter_channels = ConfigOption('scanner_counter_channels', [], missing='warn')
+    _scanner_voltage_ranges = ConfigOption('scanner_voltage_ranges', missing='error')
+    _scanner_position_ranges = ConfigOption('scanner_position_ranges', missing='error')
+
+    # odmr
+    _odmr_trigger_channel = ConfigOption('odmr_trigger_channel', missing='error')
+    _odmr_trigger_line = ConfigOption('odmr_trigger_line', 'Dev1/port0/line0', missing='warn')
+    _odmr_switch_line = ConfigOption('odmr_switch_line', 'Dev1/port0/line1', missing='warn')
+
+    # used as a default for expected maximum counts
+    _max_counts = ConfigOption('max_counts', default=3e7)
+    # timeout for the Read or/and write process in s
+    _RWTimeout = ConfigOption('read_write_timeout', default=10)
+    _counting_edge_rising = ConfigOption('counting_edge_rising', default=True)
+
+    def on_activate(self):
+        """ Starts up the NI Card at activation.
+        """
+        # the tasks used on that hardware device:
+        self._counter_daq_tasks = []
+        self._counter_analog_daq_task = None
+        self._clock_daq_task = None
```

Comment:
```suggestion
```
These are not needed
self._counter_daq_tasks = []
self._counter_analog_daq_task = None
self._clock_daq_task = None